app/src/Pages/AddContribution/AddContribution.js

```diff
@@ -0,0 +1,32 @@
+import React, { Component } from "react";
+import PageHoc from "../../components/PageHoc/PageHoc";
+import AddContributionForm from "../../components/Forms/AddContribution/index"
+import { connect } from "react-redux";
+import { login } from "../../state/ducks/auth";
+
+class AddContribution extends Component {
+  componentWillUpdate(newprops){
+    if (!(typeof newprops.state.me == 'undefined' || !newprops.state.me)) {
+      this.props.history.push('/dashboard')
+    }
+  }
+  render() {
+    return (
+      <PageHoc>
+        <AddContributionForm {...this.props} />
+      </PageHoc>
+    );
+  }
+}
+
+export default connect(
```

Comment:
When you combine the pages, this should be pulled out into `app/src/Pages/Portal/Contributions/index.js` and then do something like this:
```js
import { connect } from "react-redux";
import TopNavigation from "./Contributions";
import { isLoggedIn } from "../../state/ducks/auth";

export default connect(
state => { 
    return {state: state.auth}
  }, 
  dispatch => {
    return {
      login: (email,password) => dispatch(login(email,password)),
      dispatch
      }
    }
)(Contributions);

```

Reply:
Ok, assuming `TopNavigation` should be `Contributions`.